### PR TITLE
Feature/SK-553 | Add pagination option to REST-API

### PR DESCRIPTION
**Pull request by niklastheman**
Extended the api server to accept limit and skip params enabling pagination.

### CODE REVIEW

File: fedn/fedn/network/statestore/mongostatestore.py [hunk 1]
```diff
@@ -76,105 +76,153 @@ def get_config(self):
...
+            print(
+                "Not updating state, already in {}".format(
+                    ReducerStateToString(state)
+                )
+            )
 
-    def get_sessions(self):
-        """ Get all sessions.
+    def get_sessions(self, limit=None, skip=None, sort_key="_id", sort_order=pymongo.DESCENDING):
+        """Get all sessions.
```
Comment (Wrede) [inline fedn/fedn/network/statestore/mongostatestore.py:115]: missing args in doc string

File: fedn/fedn/network/statestore/mongostatestore.py [hunk 2]
```diff
@@ -355,96 +508,180 @@ def get_reducer(self):
...
+            ret = self.clients.find({"key": name})
             if list(ret) == []:
                 return None
             else:
                 return ret
         except Exception:
             return None
 
-    def list_clients(self):
+    def list_clients(self, limit=None, skip=None, status=False, sort_key="last_seen", sort_order=pymongo.DESCENDING):
```
Comment (Wrede) [inline fedn/fedn/network/statestore/mongostatestore.py:619]: missing args in doc string

File: fedn/fedn/network/statestore/mongostatestore.py [hunk 3]
```diff
@@ -355,96 +508,180 @@ def get_reducer(self):
...
+            result = self.clients.aggregate(pipeline)
+
+        except Exception as e:
+            print("ERROR: {}".format(e), flush=True)
+
+        return result
 
     def update_client_status(self, client_data, status, role):
-        """ Set or update client status.
+        """Set or update client status.
```
Comment (Wrede) [inline fedn/fedn/network/statestore/mongostatestore.py:684]: missing status arg in doc string

**Review by Wrede — CHANGES_REQUESTED** [review]

File: fedn/fedn/network/statestore/mongostatestore.py (hunk 1 above)
Comment (niklastheman) [reply fedn/fedn/network/statestore/mongostatestore.py:115]: done

File: fedn/fedn/network/statestore/mongostatestore.py (hunk 3 above)
Comment (niklastheman) [reply fedn/fedn/network/statestore/mongostatestore.py:696]: done

**Review by Wrede — APPROVED** [review]

**Pull request merged**